### PR TITLE
OpenStack migration using the UI

### PR DESCRIPTION
**Pull request by RichardHoch**
MTV 2.4

Partially resolves https://issues.redhat.com/browse/MTV-426 by describing how to define OpenStack source providers using the UI.

Also includes changes made to the procedures for defining other providers in the UI.

Previews:

1. Adding source providers (vmWare, RHV, and OpenStack): http://file.emea.redhat.com/rhoch/openstack_ui/html-single/#adding-source-providers  [Sorry I cannot break the file down to each of the source procviders]

2: Adding destination providers (OpenShift Virtualization): http://file.emea.redhat.com/rhoch/openstack_cli/html-single/#adding-destination-providers

### CODE REVIEW

**Review by sgratch — COMMENTED** [review]
LGTM, few minor comments.

File: documentation/modules/adding-virt-provider.adoc [hunk 1]
```diff
@@ -17,13 +17,14 @@ You can add {a-virt} destination provider to the {project-short} web console in
...
 . Select *{virt}* from the *Type* list.
-. Complete the following fields:
+. Fill in the following fields:
 
-* *Cluster name*: Specify the cluster name to display in the list of target providers.
+* *Name*: Specify the cluster name to display in the list of target providers.
 * *URL*: Specify the API endpoint of the cluster.
 * *Service account token*: Specify the `cluster-admin` service account token.
++
+If both *URL* and *Service account token* are left blank, the local cluster provider is used.
```
Comment (ahadas) [inline documentation/modules/adding-virt-provider.adoc:26]:
```suggestion
If both *URL* and *Service account token* are left blank, the local openshift cluster is used.
```

File: documentation/modules/osh-adding-source-provider.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+====
+
+[NOTE]
+====
+Migration using {osh} source providers only supports VMs whose volume snapshots are supported by {osh} Block Storage (Cinder). Migrations for VMs that use {osh} Swift at all or use only {osh} Glance for backend storage are not currently supported.
+====
+
+.Prerequisites
+
+* {manager} CA certificate or {manager} Apache CA certificate if the {manager} CA certificate was replaced by a third-party certificate on the Apache server
```
[the remark this answers is not in the document]
Comment (ahadas) [reply documentation/modules/osh-adding-source-provider.adoc:29]: right, I also commented on that in #360

[the remark this answers is not in the document]
Comment (RichardHoch) [reply documentation/modules/osh-adding-source-provider.adoc:29]: Removing this prerequisite based on chat i had with Arik Thurs. afternoon. Let me know if anyone objects to me removing it.

**Comment by RichardHoch** [comment]
@ahadas Please review again.

**Comment by RichardHoch** [comment]
@ahadas @qiyann Please review this PR.

File: documentation/modules/adding-source-provider.adoc [hunk 3]
```diff
@@ -19,8 +19,7 @@ You can add {a-rhv} source provider by using the {project-short} web console.
 
 .Prerequisites
 
-* CA certificate of the {manager}.
-* Apache CA certificate of the {manager}, in case the {manager} Apache CA certificate was replaced with a third-party CA certificate.
+*  {manager} CA certificate or {manager} Apache CA certificate if the {manager} CA certificate was replaced by a third-party certificate on the Apache server.
```
Comment (ahadas) [inline documentation/modules/adding-source-provider.adoc:22]: the phrasing in #354 (`{manager} CA certificate, unless it was replaced by a third-party certificate, in which case, specify the {manager} Apache CA certificate`) is better - need to make sure this change won't override it

File: documentation/modules/adding-virt-provider.adoc [hunk 4]
```diff
@@ -17,13 +17,14 @@ You can add {a-virt} destination provider to the {project-short} web console in
...
 . Select *{virt}* from the *Type* list.
-. Complete the following fields:
+. Fill in the following fields:
 
-* *Cluster name*: Specify the cluster name to display in the list of target providers.
+* *Name*: Specify the provider name to display in the list of target providers.
 * *URL*: Specify the API endpoint of the cluster.
 * *Service account token*: Specify the `cluster-admin` service account token.
++
+If both *URL* and *Service account token* are left blank, the local OpenShift cluster is used.
```
Comment (ahadas) [inline documentation/modules/adding-virt-provider.adoc:26]:
```suggestion
If both *URL* and *Service account token* are left blank, the local {ocp} cluster is used.
```

File: documentation/modules/osh-adding-source-provider.adoc [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+====
+
+[NOTE]
+====
+Migration using {osp} source providers only supports VMs that use only Cinder volumes.
+====
+
+.Procedure
+
+. In the {project-short} web console, click *Providers*.
```
Comment (ahadas) [inline documentation/modules/osh-adding-source-provider.adoc:29]: this needs to align with the changes in #357 (also above where `{project-short} web console` appears)

File: documentation/modules/osh-adding-source-provider.adoc [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+. Click *Add provider*.
+
+. Select *Red Hat OpenStack Platform* from the *Type* list.
+. Fill in the following fields:
+
+* *Name*: Name to display in the list of providers
+* *{osh} Identity (Keystone) URL*: {osh} Identity (Keystone) endpoint, for example, `http://controller:5000/v3`
+* *{osh} Identity (Keystone) username*: For example, `admin`
+* *{osh} Identity (Keystone) password*:
+* *Domanin*:
```
Comment (ahadas) [inline documentation/modules/osh-adding-source-provider.adoc:39]:
```suggestion
* *Domain*:
```

File: documentation/modules/adding-virt-provider.adoc [hunk 7]
```diff
@@ -17,13 +17,14 @@ You can add {a-virt} destination provider to the {project-short} web console in
 . In the  {project-short}  web console, click *Providers*.
 . Click *Add provider*.
 . Select *{virt}* from the *Type* list.
-. Complete the following fields:
+. Fill in the following fields:
 
-* *Cluster name*: Specify the cluster name to display in the list of target providers.
+* *Name*: Specify the provider name to display in the list of target providers.
 * *URL*: Specify the API endpoint of the cluster.
```
Comment (qiyuann) [inline documentation/modules/adding-virt-provider.adoc:23]:
```suggestion
* *Kubernetes API server URL*: Specify the API endpoint of the cluster.
```

File: documentation/modules/osh-adding-source-provider.adoc [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+[NOTE]
+====
+Migration using {osp} source providers only supports VMs that use only Cinder volumes.
+====
+
+.Procedure
+
+. In the {project-short} web console, click *Providers*.
+. Click *Add provider*.
```
Comment (qiyuann) [inline documentation/modules/osh-adding-source-provider.adoc:30]:
```suggestion
. Click *Create Provider*.
```

File: documentation/modules/osh-adding-source-provider.adoc [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+. Select *Red Hat OpenStack Platform* from the *Type* list.
```
Comment (qiyuann) [inline documentation/modules/osh-adding-source-provider.adoc:32]:
```suggestion
. Select *Red Hat OpenStack Platform* from the *Provider type* list.
```

File: documentation/modules/osh-adding-source-provider.adoc [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+. Fill in the following fields:
+
+* *Name*: Name to display in the list of providers
```
Comment (qiyuann) [inline documentation/modules/osh-adding-source-provider.adoc:35]:
```suggestion
* *Provider name*: Name to display in the list of providers
```

File: documentation/modules/osh-adding-source-provider.adoc [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+* *{osh} Identity (Keystone) URL*: {osh} Identity (Keystone) endpoint, for example, `http://controller:5000/v3`
```
Comment (qiyuann) [inline documentation/modules/osh-adding-source-provider.adoc:36]:
```suggestion
* *{osh} Identity server URL*: {osh} Identity (Keystone) endpoint, for example, `http://controller:5000/v3`
```

File: documentation/modules/osh-adding-source-provider.adoc [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+* *{osh} Identity (Keystone) username*: For example, `admin`
```
Comment (qiyuann) [inline documentation/modules/osh-adding-source-provider.adoc:37]:
```suggestion
* *{osh} username*: For example, `admin`
```

File: documentation/modules/osh-adding-source-provider.adoc [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+* *{osh} Identity (Keystone) password*:
```
Comment (qiyuann) [inline documentation/modules/osh-adding-source-provider.adoc:38]:
```suggestion
* *{osh} password*:
```

**Review by qiyuann — CHANGES_REQUESTED** [review]

File: documentation/modules/osh-adding-source-provider.adoc [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+.Procedure
+
+. In the {ocp} web console, click *Migration* -> *Providers for virtualization*.
+. Click *Create provider*.
+
+. Select *Red Hat OpenStack Platform* from the *Provider type* list.
+. Fill in the following fields:
+
+* *Provider name*: Name to display in the list of providers
+* *{osp} Identity server URL*: {osh} Identity (Keystone) endpoint, for example, `http://controller:5000/v3`
```
Comment (ahadas) [inline documentation/modules/osh-adding-source-provider.adoc:36]:
```suggestion
* *{osp} Identity server URL*: {osp} Identity (Keystone) endpoint, for example, `http://controller:5000/v3`
```
?

**Review by ahadas — COMMENTED** [review]
minor comment ({osh} -> {osp}), other than that - lgtm

File: documentation/modules/adding-source-provider.adoc [hunk 15]
```diff
@@ -24,31 +24,34 @@ endif::[]
 
 .Procedure
 
-. In the {project-short} web console, click *Providers*.
-. Click *Add provider*.
+. In the {ocp} web console, click *Migration* -> *Providers for virtualization*.
+. Click *Create provider*.
```
Comment (qiyuann) [inline documentation/modules/adding-source-provider.adoc:28]:
```suggestion
. Click *Create Provider*.
```

File: documentation/modules/adding-source-provider.adoc [hunk 16]
```diff
@@ -24,31 +24,34 @@ endif::[]
...
 * *VDDK init image*: VDDKInitImage path
-* Click *Verify certificate*.
-* Select the *I trust the authenticity of this certificate* checkbox.
+
+. To allow a migration without validating the provider's CA certificate, select the *Skip certificate validation* check box. By default, the checkbox is cleared, meaning that the certificate will be validated.
+. Enter the *SHA-1 fingerprint*.
 endif::[]
 ifdef::rhv[]
-. Select *Red Hat Virtualization* from the *Type* list.
+. Select *Red Hat Virtualization* from the *Provider Type* list.
```
Comment (qiyuann) [inline documentation/modules/adding-source-provider.adoc:43]:
```suggestion
. Select *Red Hat Virtualization* from the *Provider type* list.
```

File: documentation/modules/adding-virt-provider.adoc [hunk 17]
```diff
@@ -6,24 +6,25 @@
...
 * You must have {a-virt} link:https://access.redhat.com/documentation/en-us/openshift_container_platform/{ocp-version}/html/authentication_and_authorization/using-service-accounts[service account token] with `cluster-admin` privileges.
 
 .Procedure
 
-. In the  {project-short}  web console, click *Providers*.
-. Click *Add provider*.
-. Select *{virt}* from the *Type* list.
-. Complete the following fields:
+. In the {ocp} web console, click *Migration* -> *Providers for virtualization*.
+. Click *Create provider*.
```
Comment (qiyuann) [inline documentation/modules/adding-virt-provider.adoc:18]:
```suggestion
. Click *Create Provider*.
```

**Comment by RichardHoch** [comment]
@ahadas Fixed it.
@qiyuann  I made the changes you suggested. Please check the PR again. Thanks!

**Review by qiyuann — APPROVED** [review]
LGTM

**Pull request merged**